From 3944df1bd24b7295180d33dba9ee7d531c1dddd4 Mon Sep 17 00:00:00 2001
From: Maxim Ermilov <zaspire@rambler.ru>
Date: Tue, 8 Mar 2011 20:15:20 +0300
Subject: [PATCH] altTab: fix incorrect positioning with multiple monitors

Calculate the position of the alt-tab popup correctly when primary.x != 0.
This was accidentally broken by 614176b2691981c982e77b8a04b01ff02a670857
https://bugzilla.gnome.org/show_bug.cgi?id=644206
---
 js/ui/altTab.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/ui/altTab.js b/js/ui/altTab.js
index 44fd39d..46ab9a3 100644
--- a/js/ui/altTab.js
+++ b/js/ui/altTab.js
@@ -87,7 +87,7 @@ AltTabPopup.prototype = {
         let [childMinHeight, childNaturalHeight] = this._appSwitcher.actor.get_preferred_height(primary.width - hPadding);
         let [childMinWidth, childNaturalWidth] = this._appSwitcher.actor.get_preferred_width(childNaturalHeight);
         childBox.x1 = Math.max(primary.x + leftPadding, primary.x + Math.floor((primary.width - childNaturalWidth) / 2));
-        childBox.x2 = Math.min(primary.width - hPadding, childBox.x1 + childNaturalWidth);
+        childBox.x2 = Math.min(primary.x + primary.width - hPadding, childBox.x1 + childNaturalWidth);
         childBox.y1 = primary.y + Math.floor((primary.height - childNaturalHeight) / 2);
         childBox.y2 = childBox.y1 + childNaturalHeight;
         this._appSwitcher.actor.allocate(childBox, flags);
-- 
1.7.4.1

